### PR TITLE
Third task

### PR DESCRIPTION
**Pull request by siryione**

### CODE REVIEW

File: 3_another_title/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.16)
+
+project(3_another_title)
+
+file(GLOB project_src "*.hpp" "*.cpp")
+
+add_executable(3_another_title ${project_src} )
```
Comment (yegorjke) [inline 3_another_title/CMakeLists.txt]: new line is needed

File: 3_another_title/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Answers
+## __new__ and __delete__ operators
+
+### 1. What is a memory leak? How to prevent leaking of memory?
+
+It is a situation when the memory address is no longer accessable. To prevent leaking of memory we have to release the memory after its allocation and usage and before exiting blocks.
+
+### 2. What does happen if allocating memory without providing a pointer?
+
+Memory allocation has * type and can't be converted to basic non-'*' type.
```
Comment (yegorjke) [inline 3_another_title/README.md:10]: what if the code allocates some memory but it doesn't save the address of this memory to a pointer?

File: 3_another_title/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+### 3. How to prevent throwing the "bad_alloc" exception? Provide an example.
+
+Exceptions are prevented by mechanisms of exceptions handling: wraping the "dangerous" code with `{try catch}` or simply `throw` an exception.
+
+## Smart pointers
+
+### 1. Why is it impossible to copy one `unique_ptr` to another?
+
+It is imposible because of the nature of `unique_ptr`, that are used to provide unique pointers to objects.
```
Comment (yegorjke) [inline 3_another_title/README.md:20]: Is there some way how to assign or move the resource that the one unique pointer owns to another?

**Review by yegorjke — APPROVED** [review]

File: 3_another_title/README.md (hunk 3 above)
Comment (siryione) [reply 3_another_title/README.md:20]: *that are used to provide unique pointers at current time period. N (N >=2) unique pointers can't point to the same resource simultaneously, but the ownership can be moved (std::move or <unique_ptr>.release()) to another pointer -> pointer, that passed his ownership is no longer containing the address of that resource.

File: 3_another_title/README.md (hunk 2 above)
Comment (siryione) [reply 3_another_title/README.md:10]: It causes the leak of memory -> allocated (if it has been allocated) memory can't be reached.

**Pull request merged**